### PR TITLE
RDM-13136 - Add new sub-fields to the CaseLink base type with FTAs

### PR DESCRIPTION
**Pull request by KleoG**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-13136

### Change description ###
Add new sub-fields to the CaseLink base type

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+INSERT INTO public.field_type (created_at, reference, base_field_type_id, collection_field_type_id, version)
+VALUES (now(), 'ReasonForLinkList',
+        (select id from field_type where reference = 'Collection' and version = 1 and jurisdiction_id is null),
+        (select id from field_type where reference = 'LinkReason' and version = 1 and jurisdiction_id is null), '1');
+
+insert into complex_field (reference, label, security_classification, field_type_id, complex_field_type_id)
```
Comment (sullivangareth) [inline repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql:6]: the other commands refer to `public.complex_field` - no biggie, just consistency

**Review by sullivangareth — APPROVED** [review]

File: repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql (hunk 1 above)
Comment (KleoG) [reply repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql]: Done in [d6abb84](https://github.com/hmcts/ccd-definition-store-api/pull/1180/commits/d6abb8451c9765671654a5e646b9d6a6cd493184)

File: aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json [hunk 2]
```diff
@@ -0,0 +1,3171 @@
...
+  "request": {
+    "pathVariables": {
+      "id": "FT_MasterCaseType"
+    }
+  },
+
+  "expectedResponse": {
+    "body": {
+      "id": "FT_MasterCaseType",
+      "case_fields": [
```
Comment (mattnayler) [inline aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json:18]: Can we add an `__ordering__` statement just to be on the safe side as `DEFAULT_COLLECTION_ASSERTION_MODE` isn't set on def store.

```
{
   "__ordering__": "UNORDERED",
   "__elementId__": "id"
 },
```

Repeat for any other lists with more than one value in this response.

File: aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json [hunk 3]
```diff
@@ -0,0 +1,3171 @@
...
+          "display_context_parameter": null,
+          "retain_hidden_value": null
+        },
+        {
+          "metadata": false,
+          "id": "TextAreaField",
+          "case_type_id": "FT_MasterCaseType",
+          "label": "Text Area",
+          "hint_text": null,
+          "field_type": {
```
Comment (mattnayler) [inline aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json:60]: Feel free to set some of the more complex properties to `"[[ANYTHING_PRESENT]]"` on those extra fields we are not reviewing in this test.  e.g. `field_type`, `acls` on fields that are not `CaseLink` or `LinkReason`.

**Review by mattnayler — COMMENTED** [review]
See minor comments.

File: aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json (hunk 2 above)
Comment (KleoG) [reply aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json:18]: Done in [bcc54b2](https://github.com/hmcts/ccd-definition-store-api/pull/1180/commits/bcc54b212c1c01e12253ae4449ee470f6bd6f611)

File: aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json (hunk 3 above)
Comment (KleoG) [reply aat/src/aat/resources/features/F-107 CaseLink/S-107.1_Get_CaseType.td.json]: Done in [bcc54b2](https://github.com/hmcts/ccd-definition-store-api/pull/1180/commits/bcc54b212c1c01e12253ae4449ee470f6bd6f611)

File: repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
+INSERT INTO public.field_type (created_at, reference, base_field_type_id, collection_field_type_id, version)
```
[the remark this answers is not in the document]
Comment (KleoG) [reply repository/src/main/resources/db/migration/V20220314_13136__RDM-13136-add_new_subfields_to_LinkReason.sql]: Done in [bcc54b2](https://github.com/hmcts/ccd-definition-store-api/pull/1180/commits/bcc54b212c1c01e12253ae4449ee470f6bd6f611)

**Comment by KleoG** [comment]
PR closed due to corrupt jenkins build - new [PR found here](https://github.com/hmcts/ccd-definition-store-api/pull/1194)

**Pull request closed without merging**